README.md

```diff
@@ -135,10 +135,14 @@ The following are the options that can be passed to Multer.
 
 Key | Description
 --- | ---
-`dest` or `storage` | Where to store the files
-`fileFilter` | Function to control which files are accepted
-`limits` | Limits of the uploaded data
-`preservePath` | Keep the full path of files instead of just the base name
+`dest` or `storage` | Where to store the files.
+`fileFilter` | Function to control which files are accepted.
+`limits` | Limits of the uploaded data. **Defaults:** check [busboy's page](https://github.com/mscdex/busboy#exports)
+`preservePath` | Keep the full path of files instead of just the base name. **Default:** check [busboy's page](https://github.com/mscdex/busboy#exports)
+`charset` | Character set to use when one isn't defined. Sets busboy's option `defCharset`. **Default:** `utf-8`. List of [available charsets](https://github.com/mscdex/busboy/blob/master/lib/utils.js#L384)
+`paramCharset` | For multipart forms, the default character set to use for values of part header parameters (e.g. filename) that are not extended parameters (that contain an explicit charset). Sets busboy's option `defParamCharset`. **Default:** `utf-8`. List of [available charsets](https://github.com/mscdex/busboy/blob/master/lib/utils.js#L384).
```

Comment:
Shouldn't the default charset be `latin1` if the existing default is `latin1` ? 

You are trying to override `latin1` to be `utf-8` correct?

Reply:
> You are trying to override latin1 to be utf-8 correct ( in your case ) ?

Yes, and the reason for that is that Busboy has parsed filenames as utf8 before this commit 3 months ago https://github.com/mscdex/busboy/commit/d7e4e2dfc4c98c72ec9d495b38ba38c2219bfe87

My opinion is that Multer should set its own defaults to ensure consistent behaviour and not rely on busboys defaults, because those may change without notice. This could be dealt with fixed version pinnings in package.json, but that may not be the best option.

Also `preservePath` should be set to `false` by Multer to ensure consistent behaviour. Although with that parameter I don't think that Busboy would change it to be `true`. Reasoning for this is that it depend on the client whether or not it sends the path of the file or not. And most of the clients don't send it.

And for the parameter handling; I didn't what to use same parameter names as Busboy because I see that Multer could someday relay on some other solution than Busboy and for example `highwatermark` params were named inconsistently in Busboy.

Reply:
Busboy currently is not interested if `filename` field is said to be utf-8, it still uses `latin1` decoder at least in Busboy 1.6.0.

Here Busboy [takes filename](https://github.com/mscdex/busboy/blob/master/lib/types/multipart.js#L316-L323) from multipart form. It does not matter whether or not `filename` has encoding set like this `filename*=utf-8` it [still uses same decoder](https://github.com/mscdex/busboy/blob/master/lib/types/multipart.js#L305-L306), that is [defined here](https://github.com/mscdex/busboy/blob/master/lib/types/multipart.js#L236-L239) and if `defParamCharset` is not given, decoder will be `latin1`

And because of this knowledge, I think that this unit test in `lts` branch is broken https://github.com/expressjs/multer/blob/lts/test/unicode.js#L37-L44 and proper fix would be configuring `defParamCharset: 'utf8'` into Busboy, either Multer provides configuration option for that charset or hardcodes one into [make-middleware.js](https://github.com/expressjs/multer/blob/lts/lib/make-middleware.js#L28)

We also tested yesterday that Chrome is not adding encoding like this `filename*=utf-8` into form. (we are using `FormData`)

I just checked and Busboy 1.4.0 has worked ok with utf-8 filenames, but after that the breaking commit is added
<img width="1255" alt="Screenshot 2022-06-03 at 7 29 31" src="https://user-images.githubusercontent.com/7492812/171786511-fc6733b9-2d42-4834-8cf9-8559bea3e7fa.png">

Reply:
Perhaps I should comment this Unicode unit test again https://github.com/expressjs/multer/blob/lts/test/unicode.js#L37-L44

Because Busboy is handling `filename*=utf-8` in a same way as `filename=` AND because Busboy >=1.4.0 parsed filenames always as utf-8; that test passed always. With Busboy 1.5.0+ that test should fail.

Reply:
Tested, it didn't fail. So I'm missing now something here. 🤔 
  
We had Multer 1.4.2 that installed Busboy 0.2.14. Filename `test-åäöèøßð.pdf` is uploaded from Chrome 100
<img width="546" alt="Screenshot 2022-06-03 at 9 45 23" src="https://user-images.githubusercontent.com/7492812/171801946-4fde1101-d883-479a-9c6c-078620126918.png">

Filename looks like that and Multer returns `req.file.originalname` as `test-åäöèøßð.pdf`

Because of this npm audit issue https://github.com/advisories/GHSA-wm7h-9275-46v2 we updated Multer to 1.4.5-lts.1 that installs Busboy 1.6.0

Now, browser sends file as it did previosly `test-åäöèøßð.pdf`, but Multer `req.file.originalname` is `test-aÌaÌoÌeÌÃ¸ÃÃ°.pdf`. This can be fixed by setting `defParamCharset = utf-8` into Busboy.